optuna/study/study.py

```diff
@@ -36,17 +36,18 @@
 from optuna.study._study_summary import StudySummary  # NOQA
 from optuna.study._tell import _tell_with_warning
 from optuna.trial import create_trial
-from optuna.trial import FrozenTrial
 from optuna.trial import TrialState
 
 
 _dataframe = _LazyImport("optuna.study._dataframe")
 
 if TYPE_CHECKING:
     from optuna.study._dataframe import pd
+    from optuna.trial import FrozenTrial
+    from optuna.trial import Trial
 
 
-ObjectiveFuncType = Callable[[trial_module.Trial], Union[float, Sequence[float]]]
+ObjectiveFuncType = Callable[["Trial"], Union[float, Sequence[float]]]
```

Comment:
How about moving it under `if TYPE_CHECKING:`?

```suggestion
    ObjectiveFuncType = Callable[[Trial], Union[float, Sequence[float]]]
```

Reply:
Oh, I did not know that it works in this way!
I reflected your suggestion!